xlators/protocol/client/src/client.c

```diff
@@ -2302,7 +2302,17 @@ client_rpc_notify(struct rpc_clnt *rpc, void *mydata, rpc_clnt_event_t event,
                 rpc_clnt_cleanup_and_start(rpc);
 
             } else {
-                rpc->conn.config.remote_port = 0;
+                int32_t remote_port = 0;
+                ret = dict_get_int32(this->options, "remote-port",
+                                     &remote_port);
+                if (!ret) {
```

Comment:
This will log a message only on success.
```suggestion
                if (ret < 0) {
```